indexer/bin/piece-indexer.js

```diff
@@ -0,0 +1,58 @@
+import assert from 'assert'
+import { Redis } from 'ioredis'
+import { walkChain } from '../lib/advertisement-walker.js'
+import { runIpniSync } from '../lib/ipni-watcher.js'
+import { RedisRepository } from '../lib/redis-repository.js'
+
+/** @import { ProviderToInfoMap } from '../lib/typings.d.ts' */
+
+const {
+  REDIS_URL: redisUrl = 'redis://localhost:6379'
+} = process.env
+
+// TODO: setup Sentry
+
+const redisUrlParsed = new URL(redisUrl)
+const redis = new Redis({
+  host: redisUrlParsed.hostname,
+  port: Number(redisUrlParsed.port),
+  username: redisUrlParsed.username,
+  password: redisUrlParsed.password,
+  lazyConnect: true, // call connect() explicitly so that we can exit on connection error
+  family: 6 // required for upstash
+})
+
+await redis.connect()
+const repository = new RedisRepository(redis)
+
+/** @type {Set<string>} */
+const providerIdsActivelyWalked = new Set()
```

Comment:
```suggestion
const providerIdsBeingWalked = new Set()
```

I think actively walked can also mean that they have been actively walked. This conveys more clearly that they are being walked